influxdb_client/client/write/retry.py

```diff
@@ -58,16 +54,10 @@ def get_backoff_time(self):
         if consecutive_errors_len < 0:
             return 0
 
-        backoff_value = self.backoff_factor * (self.exponential_base ** consecutive_errors_len) + self._jitter_delay()
+        # Full Jitter strategy
+        backoff_value = self.backoff_factor * (self.exponential_base ** consecutive_errors_len) * self._random()
```

Comment:
I propose to change the implementation to compute the next retry delay this way:

```python
def nextDelay(attempt /* 1 means called for the first time */, options):
  range = options.first_retry_range
  i = 1
  while i<attempt:
    i++ 
    range = range * options.exponential_base
    if range >= options.max_retry_delay :
      range = options.max_retry_delay
      break
  delay = range * random()
  delay = max(options.min_retry_delay, delay) /* at least min_retry_delay */
```
Additionally, the implementation must ensure that the request is scheduled for retries after
options.max_retry_time / elapsed.

 `options.max_retry_time` can be the only meaningful configurable value from the library user POV. Settings this value to 0 can disable retry

These could be the defaults: 
```
options.first_retry_range = 5 seconds
options.exponential_base = 2
options.max_retry_delay = 125 seconds
options.min_retry_delay = 1 second
options.max_retry_time = 180 seconds
```

Reply:
This delay function does no guarantee that delay is increasing. If generated random is a lot smaller than in previous attempt, then resulting delay is also smaller. I have no smart proposal how to fix at this moment though :(

Reply:
I propose this simple modification to the above algorithm to ensure delay values are increasing
```diff
+ def randomArbitrary(min, max) {
+  return random() * (max - min) + min;
+ }
...
- delay = options.min_retry_delay + (range - options.min_retry_delay) * random() /* at least min_retry_delay */
+ delay = options.min_retry_delay + (range - options.min_retry_delay) * options.random /* at least min_retry_delay */
...
+ options.random = randomArbitrary(0.5, 1.0)
```

Reply:
Anyway, either in alg in this PR, or proposed by sranka, I think `random` needs to be "controlled" somehow, in order to avoid delay dropping when generated random is way smaller than in previous retry.

Reply:
Full Jitter is algorithm is described in https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/